src/main/java/internet/shop/web/filters/AuthorizationFilter.java

```diff
@@ -0,0 +1,74 @@
+package internet.shop.web.filters;
+
+import internet.shop.controller.user.LoginController;
+import internet.shop.lib.Injector;
+import internet.shop.model.Role;
+import internet.shop.model.User;
+import internet.shop.service.UserService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthorizationFilter implements Filter {
+    private static final Map<String, Set<Role.RoleName>> protectedUrls = new HashMap<>();
+    private static final Injector injector = Injector.getInstance("internet.shop");
+    private UserService userService
+            = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/users/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/add", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/manage", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/shopping-cart/complete-order", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/user/orders/all", Set.of(Role.RoleName.USER));
```

Comment:
what about CRUD operations of orders, users, products; shopping cart ?

Reply:
done)